modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/MeResourceManagerTest.java

```diff
@@ -0,0 +1,1373 @@
+                "  \"password\": \"kim123\",\n" +
+                "  \"id\": \"123\",\n" +
+                "  \"emails\": [\n" +
+                "      {\n" +
+                "        \"type\": \"home\",\n" +
+                "        \"value\": \"john@gmail.com\",\n" +
+                "         \"primary\": true\n" +
+                "      }\n" +
+                "  ]\n" +
+                "}";
+    }
+
+    private String getNewUserSCIMStringObjectForPATCHUpdateReplace() {
+
+        return "{\n" +
+                "  \"schemas\": \n" +
+                "    [\"urn:ietf:params:scim:schemas:core:2.0:User\",\n" +
+                "    \"urn:ietf:params:scim:schemas:extension:enterprise:2.0:User\",\n" +
+                "    \"urn:ietf:params:scim:api:messages:2.0:PatchOp\"\n" +
+                "    ],\n" +
+                "\"meta\": {\n" +
+                "    \"created\": \"2018-08-17T10:34:29Z\",\n" +
+                "    \"location\": \"https://localhost:9443/scim2/Me/008bba85-451d-414b-87de-c03b5a1f4217\",\n" +
+                "    \"lastModified\": \"2018-08-17T10:34:29Z\",\n" +
+                "    \"resourceType\": \"User\"\n" +
+                "},\n" +
+                "  \"Operations\": [\n" +
+                "    {\n" +
+                "      \"op\": \"replace\",\n" + //REPLACE Operation
+                "      \"value\": {\n" +
+                "        \"nickName\": \"shaggy\"\n" +
+                "      }\n" +
+                "    }\n" +
+                "  ],   \n" +
+                " \"name\": {\n" +
+                "    \"givenName\": \"Kim\",\n" +
+                "    \"familyName\": \"Berry\",\n" +
+                "    \"formatted\": \"Kim Berry\"\n" +
+                "  },\n" +
+                "  \"userName\": \"kimjohn  \",\n" +
+                "  \"password\": \"kim123\",\n" +
+                "  \"id\": \"123\",\n" +
+                "  \"emails\": [\n" +
+                "      {\n" +
+                "        \"type\": \"home\",\n" +
+                "        \"value\": \"johnnew@gmail.com\",\n" +
+                "         \"primary\": true\n" +
+                "      }\n" +
+                "  ]\n" +
+                "}";
+    }
+
+    private User getNewUser() throws BadRequestException, CharonException, InternalErrorException {
+
+        String scimObjectString = getNewUserSCIMObjectString();
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        JSONDecoder decoder = new JSONDecoder();
+
+        return decoder.decodeResource(scimObjectString, schema, new User());
+    }
+
+    @DataProvider(name = "dataForGetSuccess")
+    public Object[][] dataToGetSuccess() throws CharonException, InternalErrorException, BadRequestException {
+
+        User user = getNewUser();
+        String id = user.getId();
+        String name = user.getUserName();
+
+        return new Object[][]{
+                {id, name, "userName", null, 200, user}
```

Comment:
Do we need this data provider?
Since it's just one row of data it can be simply moved with the test. Please check all applicable places.

Reply:
Fixed in commit b3385c2ce740925ccf69bcb3792cd019b0c49b30